packages/dts-generator/src/phases/json-fixer.ts

```diff
@@ -134,6 +134,35 @@ function mergeOverlays(
   });
 }
 
+/**
+ * Does some rudimentary validation of method names.
+ * The method names are expected to be valid TypeScript identifiers.
+ * This is not a full validation, but catches errors that happened in real life and were hard to debug.
+ *
+ * @param symbols
+ */
+function validateNames(symbols: ConcreteSymbol[]) {
+  const invalid_characters = /[{}\[\]()<>]/;
```

Comment:
Why not really check for valid identifiers? Is there something else but identifiers that you want to allow?

A check could look like this one: 
* https://github.com/SAP/openui5/blob/master/lib/jsdoc/ui5/template/publish.js#L2328 (very strict, maybe too strict?)
* or like this one (the first identifier part of it): https://github.com/SAP/openui5/blob/master/src/sap.ui.core/src/sap/ui/base/BindingParser.js#L178

Reply:
Let's start with the simpler check and extend it later